base_layer/contacts/src/contacts_service/storage/database.rs

```diff
@@ -30,39 +30,13 @@ use log::*;
 use tari_common_types::tari_address::TariAddress;
 use tari_comms::peer_manager::NodeId;
 
-use crate::contacts_service::error::ContactsServiceStorageError;
+use crate::contacts_service::{
+    error::ContactsServiceStorageError,
+    types::{Contact, Message},
+};
 
 const LOG_TARGET: &str = "contacts::contacts_service::database";
 
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct Contact {
-    pub alias: String,
-    pub address: TariAddress,
-    pub node_id: NodeId,
-    pub last_seen: Option<NaiveDateTime>,
-    pub latency: Option<u32>,
-    pub favourite: bool,
-}
-
-impl Contact {
-    pub fn new(
-        alias: String,
-        address: TariAddress,
-        last_seen: Option<NaiveDateTime>,
-        latency: Option<u32>,
-        favourite: bool,
-    ) -> Self {
-        Self {
-            alias,
-            node_id: NodeId::from_key(address.public_key()),
-            address,
-            last_seen,
-            latency,
-            favourite,
-        }
-    }
-}
-
 /// This trait defines the functionality that a database backend need to provide for the Contacts Service
 pub trait ContactsBackend: Send + Sync + Clone {
```

Comment:
this is fine for now, but see: #5320 
This design has been copied over from our badly designed first backend on the base_node which uses a key-value store and not a relational db.